### PR TITLE
Revisit updated header spacing

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description

Revisit updated header spacing

Also:
* Bump OUI to 1.10.0

## Changelog
- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx [hunk 1]
```diff
@@ -172,7 +172,7 @@ export function TopNavMenu(props: TopNavMenuProps): ReactElement | null {
             return (
               <>
                 <MountPointPortal setMountPoint={setMenuMountPoint}>
-                  <EuiFlexGroup alignItems="stretch" gutterSize="s">
+                  <EuiFlexGroup alignItems="stretch" gutterSize="none">
```
Comment (SuZhou-Joe) [inline src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx:175]: Will it impact the case when the toggle is turned off?

Comment (AMoo-Miki) [reply src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx:175]: Nah. this is within the switch.

File: src/plugins/navigation/public/top_nav_menu/top_nav_control_item.tsx [hunk 2]
```diff
@@ -52,7 +52,10 @@ export function TopNavControlItem(props: TopNavControlData) {
       <EuiText className="descriptionHeaderControl" size="s">
         {props.description}
         {links?.map((linkProps) => (
-          <TopNavControlItem {...linkProps} />
+          <>
```
Comment (SuZhou-Joe) [inline src/plugins/navigation/public/top_nav_menu/top_nav_control_item.tsx:55]: Do we need to add a Fragment here?

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/navigation/public/top_nav_menu/top_nav_control_item.tsx (hunk 2 above)
Comment (AMoo-Miki) [reply src/plugins/navigation/public/top_nav_menu/top_nav_control_item.tsx:55]: nah. The empty tag is just for the comment.

File: src/core/public/chrome/ui/header/header.scss [hunk 3]
```diff
@@ -5,24 +5,70 @@
...
+  }
+}
+
+.headerGlobalNav:has(.newTopNavHeader:not(.primaryApplicationHeader)) {
+  margin-bottom: -1 * $euiSizeS;
+}
+
+.newTopNavHeaderTitle {
+  line-height: 1;
+  font-size: 2rem;
```
Comment (virajsanghvi) [inline src/core/public/chrome/ui/header/header.scss:71]: what is this for? Why aren't we using a token?

File: src/core/public/chrome/ui/header/header_controls_container.scss [hunk 4]
```diff
@@ -4,19 +4,21 @@
...
-    margin: 0 $euiSizeXS;
-  }
+  gap: $euiSizeS;
 
   &.headerDescriptionControl {
-    padding: 0 $euiSizeM;
-    color: $euiTextSubduedColor;
-  }
+    .descriptionHeaderControl {
+      max-width: $euiLegibilityMaxWidth;
```
Comment (virajsanghvi) [inline src/core/public/chrome/ui/header/header_controls_container.scss:11]: wouldn't does this not look weird if its above a callout that's full width?

**Review by virajsanghvi — APPROVED** [review]
Approving to get oui bumped. A little worried by setting an explicit font size and assume description thing is a design decision (although I suspect it may look weird in some cases)

File: src/core/public/chrome/ui/header/header.scss (hunk 3 above)
Comment (AMoo-Miki) [reply src/core/public/chrome/ui/header/header.scss:71]: This is a specific ask from UX.

File: src/core/public/chrome/ui/header/header_controls_container.scss (hunk 4 above)
Comment (AMoo-Miki) [reply src/core/public/chrome/ui/header/header_controls_container.scss:11]: Will let UX decide.

**Review by bandinib-amzn — APPROVED** [review]

**Pull request merged**